src/command_modules/azure-cli-container/azure/cli/command_modules/container/custom.py

```diff
@@ -210,6 +217,22 @@ def create_container(cmd,
     return sdk_no_wait(no_wait, container_group_client.create_or_update, resource_group_name, name, cgroup)
 
 
+def _build_identities_info(identities):
+    identities = identities or []
+    identity_type = ResourceIdentityType.none
+    if not identities or MSI_LOCAL_ID in identities:
+        identity_type = ResourceIdentityType.system_assigned
+    external_identities = [x for x in identities if x != MSI_LOCAL_ID]
+    if external_identities and identity_type == ResourceIdentityType.system_assigned:
+        identity_type = ResourceIdentityType.system_assigned_user_assigned
+    elif external_identities:
+        identity_type = ResourceIdentityType.user_assigned
+    identity = ContainerGroupIdentity(type=identity_type)
+    if external_identities:
```

Comment:
Why not construct the identity object after checking if there are external_identities?

Reply:
@samkreter let me know your thoughts on this.

Reply:
That would add an extra step of either having an if else creating the object with or without the user assigned identities or reassigning the external_identities variable to be in the correct format. This way seemed to me to be the most readable.

Reply:
That would add an extra step of either having an if else creating the object with or without the user assigned identities or reassigning the external_identities variable to be in the correct format. This way seemed to me to be the most readable.